axelor-account/src/main/java/com/axelor/apps/account/service/invoice/InvoiceTermServiceImpl.java

```diff
@@ -442,4 +444,116 @@ public void refusalToPay(
 
     invoiceTermRepo.save(invoiceTerm);
   }
+
+  @Override
+  @Transactional
+  public void retrieveEligibleTerms(PaymentSession paymentSession) {
+    List<InvoiceTerm> eligibleInvoiceTermList =
+        invoiceTermRepo
+            .all()
+            .filter(
+                "self.moveLine.move.company = :company AND self.paymentMode = :paymentMode AND self.dueDate <= :paymentDatePlusMargin AND self.dueDate >= :paymentDateMinusMargin AND (self.invoice.currency = :currency OR self.moveLine.move.currency = :currency) AND self.bankDetails != NULL"
+                    + " AND (self.moveLine != NULL AND (self.moveLine.move.partner.isCustomer = TRUE OR self.moveLine.move.partner.isSupplier = TRUE OR self.moveLine.move.partner.isEmployee = TRUE) OR self.moveLine = NULL AND (self.invoice.partner.isCustomer = TRUE OR self.invoice.partner.isSupplier = TRUE OR self.invoice.partner.isEmployee = TRUE))"
+                    + " AND ((self.moveLine != NULL AND self.moveLine.account.isRetrievedOnPaymentSession = TRUE) OR (self.moveLine = NULL AND self.invoice.partnerAccount.isRetrievedOnPaymentSession = TRUE))"
+                    + " AND (self.invoice.company.accountConfig.isManagePassedForPayment = FALSE OR ((self.invoice.company.accountConfig.isManagePassedForPayment = TRUE OR self.moveLine.move.company.accountConfig.isManagePassedForPayment = TRUE) AND (self.pfpValidateStatusSelect = 2 OR self.pfpValidateStatusSelect = 4)))"
+                    + " AND self.isPaid = FALSE AND self.amountRemaining > 0 AND self.paymentSession = NULL")
+            .bind("company", paymentSession.getCompany())
+            .bind("paymentMode", paymentSession.getPaymentMode())
+            .bind(
+                "paymentDatePlusMargin",
+                paymentSession
+                    .getPaymentDate()
+                    .plusDays(paymentSession.getPaymentMode().getDaysMarginOnPaySession()))
+            .bind(
+                "paymentDateMinusMargin",
+                paymentSession
+                    .getPaymentDate()
+                    .minusDays(paymentSession.getPaymentMode().getDaysMarginOnPaySession()))
+            .bind("currency", paymentSession.getCurrency())
+            .fetch();
+    eligibleInvoiceTermList.forEach(
+        invoiceTerm -> {
+          fillEligibleTerm(paymentSession, invoiceTerm);
+          invoiceTermRepo.save(invoiceTerm);
+          Beans.get(PaymentSessionRepository.class).save(paymentSession);
```

Comment:
This save should be unnecessary, as the invoiceTerm repo will save the paymentSession in the invoiceTerm. Please remove this line. If for any reason, the PaymentSessionRepository must be called, then use the injection in the constructor to avoid injecting a service in a loop.